scalanet/src/io/iohk/scalanet/peergroup/ControlEvent.scala

```diff
@@ -6,5 +6,5 @@ object ControlEvent {
 
   case object Initialized
 
-  case class InitializationError(message: String, cause: Throwable)
+  case class InitializationError(message: String, cause: Throwable) extends RuntimeException
```

Comment:
You should provide message and cause to `RuntimeException` (`extends RuntimeException(message, cause)`), so that these values are used for generating stack trace

Reply:
done